File: Data Structures & Algorithms/reorder-linked-list/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public void reorderList(ListNode head) {
        var slow = head;
        var fast = head;

        while (fast.next != null && fast.next.next != null) {
            slow = slow.next;
            fast = fast.next.next;
        }

        var reversed = slow.next;
        var previous = (ListNode) null;

        while(reversed != null) {
            var next = reversed.next;
            reversed.next = previous;
            previous = reversed;
            reversed = next;
        }

        reversed = previous;
        slow.next = null;

        while (head != null) {
            var aux = head.next;
            var aux2 = reversed != null ? reversed.next : null;
            head.next = reversed;
            if (reversed != null) {
                head.next.next = aux;
            }
            head = aux;
            reversed = aux2;
        }
    }
}

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:    
    def mergeKLists(self, lists: List[Optional[ListNode]]) -> Optional[ListNode]:
        res = None

        for l in lists:
            current = l
            while current != None:
                if res == None:
                    res = ListNode(current.val)
                elif res.val > current.val:
                    node = ListNode(current.val)
                    node.next = res
                    res = node
                else:
                    resCurrent = res
                    while resCurrent != None:
                        if resCurrent.next == None:
                            resCurrent.next = ListNode(current.val)
                            break
                        elif resCurrent.next.val > current.val:
                            node = ListNode(current.val)
                            node.next = resCurrent.next
                            resCurrent.next = node
                            break
                        resCurrent = resCurrent.next
                current = current.next

        return res